modules/fedimint-ln-client/src/receive.rs

```diff
@@ -60,6 +61,16 @@ impl State for LightningReceiveStateMachine {
         global_context: &DynGlobalClientContext,
     ) -> Vec<StateTransition<Self>> {
         match &self.state {
+            LightningReceiveStates::SubmittedOfferV0(offer) => {
+                warn!("Got old SubmittedOfferV0 state!");
+                // can just translate into the newer version
+                let new_offer = LightningReceiveSubmittedOffer {
+                    offer_txid: offer.offer_txid,
+                    invoice: offer.invoice.clone(),
+                    receiving_key: ReceivingKey::Personal(offer.payment_keypair),
+                };
+                new_offer.transitions(global_context)
```

Comment:
Since we're keeping the old state to not break the decoding, maybe we don't even need the db migration? I think with the db migration, this should essentially never happen.

Alternatively we could remove the `SubmittedOfferV0` enum (but keep `LightningReceiveSubmittedOfferV0`), then use the db migration to migrate from `LightningReceiveSubmittedOfferV0` to `SubmittedOffer(LightningReceiveSubmittedOffer)`. That's perhaps a tad cleaner since it doesn't require adding a new enum variant, but I'm good with this too.

Reply:
Like you said above though, having the enum variant around is convenient for the tests. I think if we wanted to delete it we'd need to use the raw db methods to write in the serialized bytes. We should probably modify our migration tests so this is easier, I don't think this is a blocker for this PR

Reply:
Yeah not totally sure what the best move is here, imo the current impl probably makes the most sense for now until there is a way to do tests where we manually write serialized bytes